// @atlaskit/dynamic-table/dist/es2019/styled/empty-body.js
/* eslint-disable @repo/internal/react/require-jsdoc */
/** @jsx jsx */

import { css, jsx } from '@emotion/react';

// eslint-disable-next-line @atlaskit/design-system/no-deprecated-imports
import { gridSize } from '@atlaskit/theme/constants';
const fixedHeightStyles = css({
  height: `${gridSize() * 18}px`
});
export const EmptyViewWithFixedHeight = ({
  children,
  testId
}) => jsx("div", {
  css: fixedHeightStyles,
  "data-testid": testId && `${testId}--empty-view-with-fixed-height`
}, children);
const emptyViewContainerStyles = css({
  width: '50%',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  margin: 'auto',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  padding: '10px',
  textAlign: 'center'
});
export const EmptyViewContainer = props => {
  const {
    children,
    testId
  } = props;
  return jsx("div", {
    css: emptyViewContainerStyles,
    "data-testid": testId && `${testId}--empty-view-container`
  }, children);
};
