// @atlaskit/dynamic-table/dist/es2019/styled/loading-container-advanced.js
import _extends from "@babel/runtime/helpers/extends";
/* eslint-disable @repo/internal/react/require-jsdoc */
/** @jsx jsx */
import { forwardRef } from 'react';
import { css, jsx } from '@emotion/react';
const containerStyles = css({
  marginBottom: "var(--ds-space-300, 24px)",
  position: 'relative'
});
export const Container = /*#__PURE__*/forwardRef((props, ref) => {
  const {
    children,
    testId,
    ...rest
  } = props;
  return (
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
    jsx("div", _extends({
      css: containerStyles
    }, rest, {
      "data-testid": testId,
      ref: ref
    }), children)
  );
});
const spinnerBackdropStyles = css({
  display: 'flex',
  position: 'absolute',
  inset: 0,
  alignItems: 'center',
  justifyContent: 'center',
  pointerEvents: 'none'
});
export const SpinnerBackdrop = ({
  children,
  testId
}) => jsx("div", {
  css: spinnerBackdropStyles,
  "data-testid": testId && `${testId}--spinner-backdrop`
}, children);
const spinnerContainerStyles = css({
  position: 'relative',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  top: 0
});
export const SpinnerContainer = /*#__PURE__*/forwardRef(({
  children
}, ref) => jsx("div", {
  css: spinnerContainerStyles,
  ref: ref
}, children));
