// @atlaskit/dynamic-table/dist/es2019/styled/loading-container.js
/* eslint-disable @repo/internal/react/require-jsdoc */
/** @jsx jsx */

import { css, jsx } from '@emotion/react';
export const CSS_VAR_CONTENTS_OPACITY = '--contents-opacity';
const containerStyles = css({
  position: 'relative'
});
export const Container = ({
  children,
  testId
}) => jsx("div", {
  css: containerStyles,
  "data-testid": testId && `${testId}--container`
}, children);
const contentsContainerStyles = css({
  opacity: `var(${CSS_VAR_CONTENTS_OPACITY})`,
  pointerEvents: 'none'
});
export const ContentsContainer = ({
  children,
  contentsOpacity,
  testId
}) => jsx("div", {
  style: {
    [CSS_VAR_CONTENTS_OPACITY]: contentsOpacity
  },
  css: [contentsContainerStyles],
  "data-testid": testId && `${testId}--contents--container`
}, children);
const spinnerContainerStyles = css({
  display: 'flex',
  position: 'absolute',
  inset: 0,
  alignItems: 'center',
  justifyContent: 'center'
});
export const SpinnerContainer = ({
  children,
  testId
}) => jsx("div", {
  css: spinnerContainerStyles,
  "data-testid": testId && `${testId}--spinner--container`
}, children);
