// @atlaskit/progress-tracker/dist/esm/internal/bar.js
/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import { B300 } from '@atlaskit/theme/colors';
import { spacing } from '../constants';
import { HALF_GRID_SIZE, LABEL_TOP_SPACING, PROGRESS_BAR_HEIGHT, varSpacing, varTransitionDelay, varTransitionEasing, varTransitionSpeed } from './constants';
var progressBarStyles = css({
  height: PROGRESS_BAR_HEIGHT,
  position: 'absolute',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  left: '50%',
  backgroundColor: "var(--ds-background-brand-bold, ".concat(B300, ")"),
  borderBottomRightRadius: PROGRESS_BAR_HEIGHT,
  borderTopRightRadius: PROGRESS_BAR_HEIGHT,
  transform: "translate(0, calc(-1 * ".concat(LABEL_TOP_SPACING, "))"),
  transition: "width var(".concat(varTransitionSpeed, ") var(").concat(varTransitionEasing, ")"),
  transitionDelay: "var(".concat(varTransitionDelay, ")")
});

/**
 * __Progress bar__
 *
 * A progress bar describes the horizontal tracking bar that traverses each individual step.
 *
 */
var ProgressBar = function ProgressBar(_ref) {
  var percentageComplete = _ref.percentageComplete,
    testId = _ref.testId;
  return jsx("div", {
    "data-testid": testId,
    style: {
      width: "calc(".concat(percentageComplete, "% + ").concat(percentageComplete / 100, " * calc(var(").concat(varSpacing, ", ").concat(spacing.cosy, ") + ").concat(HALF_GRID_SIZE, "))")
    },
    css: progressBarStyles
  });
};
export default ProgressBar;
