// @atlaskit/progress-tracker/dist/esm/internal/marker.js
/* eslint-disable @atlaskit/design-system/no-nested-styles */
/** @jsx jsx */

import { css, jsx } from '@emotion/react';
import { LABEL_TOP_SPACING, PROGRESS_BAR_HEIGHT, varBackgroundColor, varMarkerColor, varTransitionDelay, varTransitionEasing, varTransitionSpeed } from './constants';
var progressMarkerStyles = css({
  width: PROGRESS_BAR_HEIGHT,
  height: PROGRESS_BAR_HEIGHT,
  position: 'absolute',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  left: '50%',
  backgroundColor: "var(".concat(varBackgroundColor, ")"),
  borderRadius: PROGRESS_BAR_HEIGHT,
  transform: "translate(-50%, calc(-1 * ".concat(LABEL_TOP_SPACING, "))"),
  transition: "opacity var(".concat(varTransitionSpeed, ") var(").concat(varTransitionEasing, "), background-color var(").concat(varTransitionSpeed, ") var(").concat(varTransitionEasing, ")"),
  transitionDelay: "var(".concat(varTransitionDelay, ")"),
  '&.fade-appear': {
    opacity: 0.01
  },
  '&.fade-appear.fade-appear-active': {
    opacity: 1
  },
  '&.fade-enter': {
    backgroundColor: "var(".concat(varMarkerColor, ")")
  },
  '&.fade-enter.fade-enter-active': {
    backgroundColor: "var(".concat(varBackgroundColor, ")")
  }
});

/**
 * __Progress marker__
 *
 * Similar to `@atlaskit/progress-indicator`, a small visual circle marker
 */
var ProgressMarker = function ProgressMarker(_ref) {
  var testId = _ref.testId;
  return (
    // too many props that would go in UNSAFE_style + css transition prop having issues

    jsx("div", {
      "data-testid": testId,
      css: progressMarkerStyles
    })
  );
};
export default ProgressMarker;
