// @atlaskit/range/dist/cjs/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Input = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("react");
var _react2 = require("@emotion/react");
var theme = _interopRequireWildcard(require("./theme"));
var _excluded = ["valuePercent", "style"];
var _hoverNotDisabled, _css;
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; } /** @jsx jsx */
var VAR_THUMB_BORDER_COLOR = '--thumb-border';
var VAR_THUMB_SHADOW = '--thumb-shadow';
var VAR_THUMB_BACKGROUND_COLOR = '--thumb-bg';
var VAR_TRACK_BACKGROUND_COLOR = '--track-bg';
var VAR_TRACK_FOREGROUND_COLOR = '--track-fg';
var VAR_TRACK_FOREGROUND_WIDTH = '--track-fg-width';
var sliderThumbStyles = {
  boxSizing: 'border-box',
  width: theme.thumb.size,
  height: theme.thumb.size,
  border: 'none',
  background: "var(".concat(VAR_THUMB_BACKGROUND_COLOR, ", ").concat(theme.thumb.background.default, ")"),
  // adapted from @atlaskit/focus-ring
  outline: "solid 2px var(".concat(VAR_THUMB_BORDER_COLOR, ")"),
  outlineOffset: '2px',
  borderRadius: "var(--ds-border-radius-circle, 50%)",
  boxShadow: "var(".concat(VAR_THUMB_SHADOW, ")"),
  cursor: 'pointer',
  // Different implicit behavior across browsers -> making it explicit.
  transition: "background-color ".concat(theme.transitionDuration, " ease-in-out")
};
var sliderTrackStyles = {
  borderRadius: theme.track.borderRadius,
  border: 0,
  cursor: 'pointer',
  height: theme.track.height,
  width: '100%',
  transition: "background-color ".concat(theme.transitionDuration, " ease-in-out")
};

// Styles are split per browser as they are implemented differently
// Cannot consolidate as Chrome & Firefox don't recognise styles if they are grouped
// with CSS selectors they don't recognise, e.g. &::-ms-thumb
var browserStyles = {
  webkit: (0, _react2.css)({
    WebkitAppearance: 'none',
    // Hides the slider so that custom slider can be made
    '::-webkit-slider-thumb': _objectSpread(_objectSpread({}, sliderThumbStyles), {}, {
      // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
      marginTop: -(theme.thumb.size - theme.track.height) / 2,
      WebkitAppearance: 'none'
    }),
    '::-webkit-slider-runnable-track': _objectSpread(_objectSpread({}, sliderTrackStyles), {}, {
      /**
       * Webkit does not have separate properties for the background/foreground like firefox.
       * Instead we use background layering:
       * - The gray background is a simple background color.
       * - The blue foreground is a 'gradient' (to create a color block) that is sized to the progress.
       *
       * Individual properties have been used over the `background` shorthand for readability.
       */
      backgroundColor: "var(".concat(VAR_TRACK_BACKGROUND_COLOR, ")"),
      backgroundImage: "linear-gradient(var(".concat(VAR_TRACK_FOREGROUND_COLOR, "), var(").concat(VAR_TRACK_FOREGROUND_COLOR, "))"),
      backgroundRepeat: 'no-repeat',
      backgroundSize: "var(".concat(VAR_TRACK_FOREGROUND_WIDTH, ") 100%"),
      // eslint-disable-next-line @atlaskit/design-system/no-nested-styles
      '[dir="rtl"] &': {
        backgroundPosition: 'right'
      }
    }),
    ':disabled': {
      '::-webkit-slider-thumb, ::-webkit-slider-runnable-track': {
        cursor: 'not-allowed'
      }
    }
  }),
  firefox: (0, _react2.css)({
    '::-moz-range-thumb': sliderThumbStyles,
    '::-moz-focus-outer': {
      border: 0
    },
    '::-moz-range-progress': _objectSpread(_objectSpread({}, sliderTrackStyles), {}, {
      background: "var(".concat(VAR_TRACK_FOREGROUND_COLOR, ")")
    }),
    '::-moz-range-track': _objectSpread(_objectSpread({}, sliderTrackStyles), {}, {
      background: "var(".concat(VAR_TRACK_BACKGROUND_COLOR, ")")
    }),
    ':disabled': {
      '::-moz-range-thumb, ::-moz-range-progress, ::-moz-range-track': {
        cursor: 'not-allowed'
      }
    }
  })
};
var baseStyles = (0, _react2.css)({
  width: '100%',
  // Has a fixed width by default
  height: theme.input.height,
  // Otherwise thumb will collide with previous box element
  background: 'transparent',
  // Otherwise white
  ':focus': {
    outline: 'none'
  },
  ':disabled': {
    cursor: 'not-allowed',
    opacity: "var(--ds-opacity-disabled, 0.4)"
  }
});
var themeStyles = (0, _react2.css)((_css = {}, (0, _defineProperty2.default)(_css, VAR_THUMB_SHADOW, theme.thumb.boxShadow.default), (0, _defineProperty2.default)(_css, VAR_TRACK_BACKGROUND_COLOR, theme.track.background.default), (0, _defineProperty2.default)(_css, VAR_TRACK_FOREGROUND_COLOR, theme.track.foreground.default), (0, _defineProperty2.default)(_css, ':hover:not(:disabled)', (_hoverNotDisabled = {}, (0, _defineProperty2.default)(_hoverNotDisabled, VAR_THUMB_BACKGROUND_COLOR, theme.thumb.background.hovered), (0, _defineProperty2.default)(_hoverNotDisabled, VAR_TRACK_BACKGROUND_COLOR, theme.track.background.hovered), (0, _defineProperty2.default)(_hoverNotDisabled, VAR_TRACK_FOREGROUND_COLOR, theme.track.foreground.hovered), _hoverNotDisabled)), (0, _defineProperty2.default)(_css, ':active:not(:disabled)', (0, _defineProperty2.default)({}, VAR_THUMB_BACKGROUND_COLOR, theme.thumb.background.pressed)), (0, _defineProperty2.default)(_css, ':focus-visible', (0, _defineProperty2.default)({}, VAR_THUMB_BORDER_COLOR, theme.thumb.borderColor.focused)), _css));

/**
 * __Input__
 * Internal-only styled input component.
 */
var Input = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var valuePercent = props.valuePercent,
    style = props.style,
    strippedProps = (0, _objectWithoutProperties2.default)(props, _excluded);
  return (0, _react2.jsx)("input", (0, _extends2.default)({}, strippedProps, {
    style: (0, _defineProperty2.default)({}, VAR_TRACK_FOREGROUND_WIDTH, "".concat(valuePercent, "%")),
    ref: ref,
    css: [baseStyles, browserStyles.webkit, browserStyles.firefox, themeStyles]
  }));
});
exports.Input = Input;
Input.displayName = 'InputRange';
