// @atlaskit/code/dist/cjs/syntax-highlighter/async.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));
var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));
var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = _interopRequireDefault(require("react"));
var _highlight = _interopRequireDefault(require("./lib/highlight"));
function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }
function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }
// Uses the loader method of async bundling
// Instantiates highligher as a singleton, loading refractor only once per page (refractor/prism are singleton modules)
var generator = function generator(options) {
  var loader = options.loader;

  // eslint-disable-next-line @repo/internal/react/no-class-components
  var AsyncHighlighter = /*#__PURE__*/function (_React$PureComponent) {
    (0, _inherits2.default)(AsyncHighlighter, _React$PureComponent);
    var _super = _createSuper(AsyncHighlighter);
    function AsyncHighlighter() {
      (0, _classCallCheck2.default)(this, AsyncHighlighter);
      return _super.apply(this, arguments);
    }
    (0, _createClass2.default)(AsyncHighlighter, [{
      key: "componentDidMount",
      value: function componentDidMount() {
        var _this = this;
        if (!AsyncHighlighter.astGeneratorPromise) {
          AsyncHighlighter.loadAstGenerator();
        }
        if (!AsyncHighlighter.astGenerator && AsyncHighlighter.astGeneratorPromise) {
          AsyncHighlighter.astGeneratorPromise.then(function () {
            _this.forceUpdate();
          });
        }
      }
    }, {
      key: "render",
      value: function render() {
        return /*#__PURE__*/_react.default.createElement(AsyncHighlighter.highlightInstance, (0, _extends2.default)({}, this.props, {
          astGenerator: AsyncHighlighter.astGenerator
        }));
      }
    }], [{
      key: "preload",
      value:
      // Useful in tests
      function preload() {
        return AsyncHighlighter.loadAstGenerator();
      }
    }, {
      key: "loadAstGenerator",
      value: function loadAstGenerator() {
        AsyncHighlighter.astGeneratorPromise = loader().then(function (astGenerator) {
          AsyncHighlighter.astGenerator = astGenerator;
          return astGenerator;
        });
        return AsyncHighlighter.astGeneratorPromise;
      }
    }]);
    return AsyncHighlighter;
  }(_react.default.PureComponent);
  (0, _defineProperty2.default)(AsyncHighlighter, "astGenerator", null);
  (0, _defineProperty2.default)(AsyncHighlighter, "highlightInstance", _highlight.default);
  return AsyncHighlighter;
};
var _default = exports.default = generator({
  loader: function loader() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require( /* webpackChunkName: "@atlaskit-internal_refractor-import" */
      'refractor'));
    }).then(function (module) {
      // Webpack 3 returns module.exports as default as module, but webpack 4 returns module.exports as module.default
      return module.default || module;
    });
  }
});
