// @atlaskit/dynamic-table/dist/esm/styled/loading-container-advanced.js
import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
var _excluded = ["children", "testId"];
/* eslint-disable @repo/internal/react/require-jsdoc */
/** @jsx jsx */
import { forwardRef } from 'react';
import { css, jsx } from '@emotion/react';
var containerStyles = css({
  marginBottom: "var(--ds-space-300, 24px)",
  position: 'relative'
});
export var Container = /*#__PURE__*/forwardRef(function (props, ref) {
  var children = props.children,
    testId = props.testId,
    rest = _objectWithoutProperties(props, _excluded);
  return (
    // eslint-disable-next-line @repo/internal/react/no-unsafe-spread-props
    jsx("div", _extends({
      css: containerStyles
    }, rest, {
      "data-testid": testId,
      ref: ref
    }), children)
  );
});
var spinnerBackdropStyles = css({
  display: 'flex',
  position: 'absolute',
  inset: 0,
  alignItems: 'center',
  justifyContent: 'center',
  pointerEvents: 'none'
});
export var SpinnerBackdrop = function SpinnerBackdrop(_ref) {
  var children = _ref.children,
    testId = _ref.testId;
  return jsx("div", {
    css: spinnerBackdropStyles,
    "data-testid": testId && "".concat(testId, "--spinner-backdrop")
  }, children);
};
var spinnerContainerStyles = css({
  position: 'relative',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  top: 0
});
export var SpinnerContainer = /*#__PURE__*/forwardRef(function (_ref2, ref) {
  var children = _ref2.children;
  return jsx("div", {
    css: spinnerContainerStyles,
    ref: ref
  }, children);
});
