// @atlaskit/primitives/dist/cjs/components/text.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _react2 = require("@emotion/react");
var _tinyInvariant = _interopRequireDefault(require("tiny-invariant"));
var _styleMaps = require("../xcss/style-maps.partial");
var _surfaceProvider = require("./internal/surface-provider");
var _excluded = ["children"];
/** @jsx jsx */
var asAllowlist = ['span', 'p', 'strong', 'em'];
// We're doing this because our CSS reset can add top margins to elements such as `p` which is totally insane.
// Long term we should remove those instances from the reset - it should be a reset to 0.
// For now, at least we know <Text> will be unaffected by this.
var resetStyles = (0, _react2.css)({
  margin: 0
});
var strongStyles = (0, _react2.css)({
  fontWeight: 'bold'
});
var emStyles = (0, _react2.css)({
  fontStyle: 'italic'
});
var textAlignMap = {
  center: (0, _react2.css)({
    textAlign: 'center'
  }),
  end: (0, _react2.css)({
    textAlign: 'end'
  }),
  start: (0, _react2.css)({
    textAlign: 'start'
  })
};
var truncationStyles = (0, _react2.css)({
  display: '-webkit-box',
  overflow: 'hidden',
  WebkitBoxOrient: 'vertical'
});
var wordBreakMap = {
  breakAll: (0, _react2.css)({
    wordBreak: 'break-all'
  })
};
var tempVariantSizeMap = {
  body: 'medium',
  'body.small': 'small',
  'body.large': 'large'
};
var HasTextAncestorContext = /*#__PURE__*/(0, _react.createContext)(false);
var useHasTextAncestor = function useHasTextAncestor() {
  return (0, _react.useContext)(HasTextAncestorContext);
};

/**
 * Custom hook designed to abstract the parsing of the color props and make it clearer in the future how color is reconciled between themes and tokens.
 */
var useColor = function useColor(colorProp, hasTextAncestor) {
  var surface = (0, _surfaceProvider.useSurface)();

  /**
   * Where the color of the surface is inverted we always override the color
   * as there is no valid choice that is not covered by the override.
   */
  if (_styleMaps.inverseColorMap.hasOwnProperty(surface)) {
    return _styleMaps.inverseColorMap[surface];
  }
  if (colorProp === 'inherit') {
    return undefined;
  }
  if (!colorProp && hasTextAncestor) {
    return undefined;
  }
  return colorProp || 'color.text';
};

/**
 * __Text__
 *
 * Text is a primitive component that has the Atlassian Design System's design guidelines baked in.
 * This includes considerations for text attributes such as color, font size, font weight, and line height.
 * It renders a `span` by default.
 *
 * @internal
 */
var Text = function Text(_ref) {
  var children = _ref.children,
    props = (0, _objectWithoutProperties2.default)(_ref, _excluded);
  var _props$as = props.as,
    Component = _props$as === void 0 ? 'span' : _props$as,
    colorProp = props.color,
    textAlign = props.textAlign,
    testId = props.testId,
    id = props.id,
    _props$size = props.size,
    size = _props$size === void 0 ? 'medium' : _props$size,
    variant = props.variant,
    weight = props.weight,
    maxLines = props.maxLines;
  (0, _tinyInvariant.default)(asAllowlist.includes(Component), "@atlaskit/primitives: Text received an invalid \"as\" value of \"".concat(Component, "\""));
  var localSize = variant && tempVariantSizeMap[variant] || size;
  var hasTextAncestor = useHasTextAncestor();
  var color = useColor(colorProp, hasTextAncestor);
  var component = (0, _react2.jsx)(Component, {
    css: [resetStyles, _styleMaps.fontStylesMap[localSize], color && _styleMaps.textColorStylesMap[color], maxLines && truncationStyles, maxLines === 1 && wordBreakMap.breakAll, textAlign && textAlignMap[textAlign], weight && _styleMaps.fontWeightStylesMap[weight], Component === 'em' && emStyles, Component === 'strong' && strongStyles],
    style: {
      WebkitLineClamp: maxLines
    },
    "data-testid": testId,
    id: id
  }, children);
  return hasTextAncestor ?
  // no need to re-apply context if the text is already wrapped
  component : (0, _react2.jsx)(HasTextAncestorContext.Provider, {
    value: true
  }, component);
};
var _default = exports.default = Text;
