// @atlaskit/dynamic-table/dist/cjs/styled/constants.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.truncationWidthStyles = exports.overflowTruncateStyles = exports.getTruncationStyleVars = exports.fixedSizeTruncateStyles = exports.cellStyles = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _react = require("@emotion/react");
var CSS_VAR_WIDTH = '--local-dynamic-table-width';
var truncationWidthStyles = exports.truncationWidthStyles = (0, _react.css)({
  width: "var(".concat(CSS_VAR_WIDTH, ")")
});
var fixedSizeTruncateStyles = exports.fixedSizeTruncateStyles = (0, _react.css)({
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap'
});
var overflowTruncateStyles = exports.overflowTruncateStyles = (0, _react.css)({
  overflow: 'hidden'
});
var getTruncationStyleVars = exports.getTruncationStyleVars = function getTruncationStyleVars(_ref) {
  var width = _ref.width;
  return typeof width !== 'undefined' ? (0, _defineProperty2.default)({}, CSS_VAR_WIDTH, "".concat(width, "%")) : undefined;
};
var cellStyles = exports.cellStyles = (0, _react.css)({
  padding: "var(--ds-space-050, 4px)".concat(" ", "var(--ds-space-100, 8px)"),
  border: 'none',
  textAlign: 'left',
  '&:first-of-type': {
    paddingLeft: "var(--ds-space-0, 0px)"
  },
  '&:last-child': {
    paddingRight: "var(--ds-space-0, 0px)"
  }
});
