// @atlaskit/dynamic-table/dist/cjs/styled/empty-body.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.EmptyViewWithFixedHeight = exports.EmptyViewContainer = void 0;
var _react = require("@emotion/react");
var _constants = require("@atlaskit/theme/constants");
/* eslint-disable @repo/internal/react/require-jsdoc */
/** @jsx jsx */

// eslint-disable-next-line @atlaskit/design-system/no-deprecated-imports

var fixedHeightStyles = (0, _react.css)({
  height: "".concat((0, _constants.gridSize)() * 18, "px")
});
var EmptyViewWithFixedHeight = exports.EmptyViewWithFixedHeight = function EmptyViewWithFixedHeight(_ref) {
  var children = _ref.children,
    testId = _ref.testId;
  return (0, _react.jsx)("div", {
    css: fixedHeightStyles,
    "data-testid": testId && "".concat(testId, "--empty-view-with-fixed-height")
  }, children);
};
var emptyViewContainerStyles = (0, _react.css)({
  width: '50%',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  margin: 'auto',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  padding: '10px',
  textAlign: 'center'
});
var EmptyViewContainer = exports.EmptyViewContainer = function EmptyViewContainer(props) {
  var children = props.children,
    testId = props.testId;
  return (0, _react.jsx)("div", {
    css: emptyViewContainerStyles,
    "data-testid": testId && "".concat(testId, "--empty-view-container")
  }, children);
};
