// @forge/react/out/uikit2-components.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequiredAsterisk = exports.ValidMessage = exports.Tooltip = exports.Toggle = exports.Textfield = exports.TextArea = exports.TagGroup = exports.Tag = exports.Tabs = exports.TabPanel = exports.TabList = exports.Tab = exports.Stack = exports.Spinner = exports.SectionMessageAction = exports.SectionMessage = exports.Select = exports.Range = exports.RadioGroup = exports.Radio = exports.ProgressTracker = exports.ProgressBar = exports.ModalTransition = exports.ModalTitle = exports.ModalHeader = exports.ModalFooter = exports.ModalBody = exports.Modal = exports.Lozenge = exports.LoadingButton = exports.LinkButton = exports.Label = exports.Inline = exports.Icon = exports.HelperMessage = exports.Heading = exports.FormSection = exports.FormHeader = exports.FormFooter = exports.Form = exports.ErrorMessage = exports.DynamicTable = exports.DatePicker = exports.CodeBlock = exports.Code = exports.Checkbox = exports.ButtonGroup = exports.Button = exports.Box = exports.Badge = void 0;
exports.Badge = 'Badge';
exports.Box = 'Box';
exports.Button = 'Button';
exports.ButtonGroup = 'ButtonGroup';
exports.Checkbox = 'Checkbox';
exports.Code = 'Code';
exports.CodeBlock = 'CodeBlock';
exports.DatePicker = 'DatePicker';
exports.DynamicTable = 'DynamicTable';
exports.ErrorMessage = 'ErrorMessage';
exports.Form = 'Form';
exports.FormFooter = 'FormFooter';
exports.FormHeader = 'FormHeader';
exports.FormSection = 'FormSection';
exports.Heading = 'Heading';
exports.HelperMessage = 'HelperMessage';
exports.Icon = 'Icon';
exports.Inline = 'Inline';
exports.Label = 'Label';
exports.LinkButton = 'LinkButton';
exports.LoadingButton = 'LoadingButton';
exports.Lozenge = 'Lozenge';
exports.Modal = 'Modal';
exports.ModalBody = 'ModalBody';
exports.ModalFooter = 'ModalFooter';
exports.ModalHeader = 'ModalHeader';
exports.ModalTitle = 'ModalTitle';
exports.ModalTransition = 'ModalTransition';
exports.ProgressBar = 'ProgressBar';
exports.ProgressTracker = 'ProgressTracker';
exports.Radio = 'Radio';
exports.RadioGroup = 'RadioGroup';
exports.Range = 'Range';
exports.Select = 'Select';
exports.SectionMessage = 'SectionMessage';
exports.SectionMessageAction = 'SectionMessageAction';
exports.Spinner = 'Spinner';
exports.Stack = 'Stack';
exports.Tab = 'Tab';
exports.TabList = 'TabList';
exports.TabPanel = 'TabPanel';
exports.Tabs = 'Tabs';
exports.Tag = 'Tag';
exports.TagGroup = 'TagGroup';
exports.TextArea = 'TextArea';
exports.Textfield = 'Textfield';
exports.Toggle = 'Toggle';
exports.Tooltip = 'Tooltip';
exports.ValidMessage = 'ValidMessage';
exports.RequiredAsterisk = 'RequiredAsterisk';
