// @atlaskit/progress-tracker/dist/cjs/internal/bar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = require("@emotion/react");
var _colors = require("@atlaskit/theme/colors");
var _constants = require("../constants");
var _constants2 = require("./constants");
/** @jsx jsx */

var progressBarStyles = (0, _react.css)({
  height: _constants2.PROGRESS_BAR_HEIGHT,
  position: 'absolute',
  // eslint-disable-next-line @atlaskit/design-system/ensure-design-token-usage
  left: '50%',
  backgroundColor: "var(--ds-background-brand-bold, ".concat(_colors.B300, ")"),
  borderBottomRightRadius: _constants2.PROGRESS_BAR_HEIGHT,
  borderTopRightRadius: _constants2.PROGRESS_BAR_HEIGHT,
  transform: "translate(0, calc(-1 * ".concat(_constants2.LABEL_TOP_SPACING, "))"),
  transition: "width var(".concat(_constants2.varTransitionSpeed, ") var(").concat(_constants2.varTransitionEasing, ")"),
  transitionDelay: "var(".concat(_constants2.varTransitionDelay, ")")
});

/**
 * __Progress bar__
 *
 * A progress bar describes the horizontal tracking bar that traverses each individual step.
 *
 */
var ProgressBar = function ProgressBar(_ref) {
  var percentageComplete = _ref.percentageComplete,
    testId = _ref.testId;
  return (0, _react.jsx)("div", {
    "data-testid": testId,
    style: {
      width: "calc(".concat(percentageComplete, "% + ").concat(percentageComplete / 100, " * calc(var(").concat(_constants2.varSpacing, ", ").concat(_constants.spacing.cosy, ") + ").concat(_constants2.HALF_GRID_SIZE, "))")
    },
    css: progressBarStyles
  });
};
var _default = exports.default = ProgressBar;
