// @forge/react/out/__test__/testUtils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.findElementInForgeDoc = exports.getLastBridgeCallForgeDoc = exports.getLastBridgeCall = exports.setupBridge = void 0;
const setupBridge = () => {
    const bridgeCalls = [];
    global['self'] = {
        __bridge: {
            callBridge: (cmd, data) => {
                bridgeCalls.push({ cmd, data });
            }
        }
    };
    return bridgeCalls;
};
exports.setupBridge = setupBridge;
const getLastBridgeCall = (bridgeCalls) => {
    if (bridgeCalls.length === 0) {
        return null;
    }
    return bridgeCalls[bridgeCalls.length - 1];
};
exports.getLastBridgeCall = getLastBridgeCall;
const getLastBridgeCallForgeDoc = (bridgeCalls) => {
    const lastBridgeCall = (0, exports.getLastBridgeCall)(bridgeCalls);
    if (lastBridgeCall === null) {
        return lastBridgeCall;
    }
    return lastBridgeCall.data.forgeDoc;
};
exports.getLastBridgeCallForgeDoc = getLastBridgeCallForgeDoc;
const findElementInForgeDoc = (forgeDoc, key) => {
    if (forgeDoc === null) {
        return;
    }
    if (forgeDoc.key === key) {
        return forgeDoc;
    }
    const { children } = forgeDoc;
    if (forgeDoc.children) {
        for (const child of children) {
            const result = (0, exports.findElementInForgeDoc)(child, key);
            if (result) {
                return result;
            }
        }
    }
    return;
};
exports.findElementInForgeDoc = findElementInForgeDoc;
